core/renderers/zelos/info.js

```diff
@@ -259,42 +246,51 @@ Blockly.zelos.RenderInfo.prototype.adjustXPosition_ = function() {
 };
 
 /**
- * @override
+ * Finalize the output connection info.  In particular set the height of the
+ * output connection to match that of the block.  For the right side, add a
+ * right connection shape element and have it match the dimensions of the
+ * output connection.
+ * @protected
  */
-Blockly.zelos.RenderInfo.prototype.finalize_ = function() {
-  // Performance note: this could be combined with the draw pass, if the time
-  // that this takes is excessive.  But it shouldn't be, because it only
-  // accesses and sets properties that already exist on the objects.
+Blockly.zelos.RenderInfo.prototype.finalizeOutputConnection_ = function() {
   var yCursor = 0;
+  var firstInpurRow = null;
+  // Determine the block height.
   for (var i = 0, row; (row = this.rows[i]); i++) {
     row.yPos = yCursor;
     yCursor += row.height;
+    if (!firstInpurRow && Blockly.blockRendering.Types.isInputRow(row)) {
+      firstInpurRow = row;
+    }
   }
-  this.height = yCursor;
-
   if (this.outputConnection && this.outputConnection.isDynamicShape) {
     // Dynamic output connections depend on the height of the block. Adjust the
     // height of the connection.
-    this.outputConnection.setShapeDimensions(
-        this.outputConnection.shape.height(this.height),
-        this.outputConnection.shape.width(this.height));
+    var connectionHeight = this.outputConnection.shape.height(yCursor);
+    var connectionWidth = this.outputConnection.shape.width(yCursor);
 
-    // Recompute the bounds as we now know the output connection dimensions.
-    this.computeBounds_();
-  }
+    this.outputConnection.height = connectionHeight;
```

Comment:
I agree that the output connection should become a measurable and this should move into it.